### ISSUE TITLE
training code?

### ISSUE THREAD

**Issue by noahzhy** [TheJaeLal/LineFormer#3]
is there a manual to train this?

**Comment by TheJaeLal** [comment]
yes, will be up soon

**Comment by CHENZHI99** [comment]
> yes, will be up soon

Could you please share when will you public the training manual?

**Comment by TheJaeLal** [comment]
Apologies for the delayed response. Unfortunately, I haven't been able to find the time to update training scripts/env due to some urgent personal commitment.

However, you can try referring to the [MMDetection docs](https://mmdetection.readthedocs.io/en/latest/user_guides/train.html) for training instructions. I'm attaching the lineformer config file that you can use for this process. 

You will have to create the dataset in ms-coco format, with object (segmentation) masks encoded using rle. (refer the sample annotation file). More info on the format can be found [here](https://cocodataset.org/#format-results)

Hope this helps. 

Attached Files

[val_coco_annot.json](https://github.com/TheJaeLal/LineFormer/files/13877498/val_coco_annot.json)
[lineformer_config.txt](https://github.com/TheJaeLal/LineFormer/files/13877522/lineformer_config.txt)

Change the '.txt' extension in config file attached here to '.py'

**Comment by sazidrj** [comment]
can you provide instructions for training?

**Comment by sazidrj** [comment]
can you provide train and test data? or how you have create data ?. It is confusing in the paper. The file 'val_coco_annot.json' you provided, all image files having same height and width. If you resize the image then how you can create mask image because {x,y} points from ground truth will change. If I resize the image after creating mask then rle encoding is not correct.

**Comment by TheJaeLal** [comment]
Hi @sazidrj. Sorry for the late response. I have added the dataset generation notebooks inside 'data_processing' directory. To answer your query, in DataFormat.ipynb, you'll see the code snippet 'generate_mask_for_clean_img', where once the images are resized for batch training (maintaining aspect ratio), the corresponding image transformation matrix is also used to transform the ground-truth x,y coordinates and from these, the mask is being generated.

**Comment by sazidrj** [comment]
Thank you for your response

**Comment by melika-sce** [comment]
Hi @sazidrj can you provide an explanation about how did you labeled your data?
I need to annotated my data but I don't know which tool should I use for labeling and how should I label them (polygon, keypoints, or any other way). I will appreciate your help since I am confused about data preparation section.

**Comment by shz0519** [comment]
Hi，I'm preparing to reproduce your LineFormer.  can you provide the pretrian_ckpt in ‘lineformer_config.txt’? mask2former_swin-t-p4-w7-224_lsj_8x2_50e_coco_20220508_151657-b201f86d.pth

**Comment by shz0519** [comment]
Hi, in DataFormat.ipynb, it mentioned "import pmc_line_loader", but I couldn't find where pmc_line_loader is located. Could you provide the file where it is?
